### ISSUE TITLE
Is there a way to "overwrap" search for next mc/mark-more-like-this ?

### ISSUE THREAD

**Issue by FelixBrendel** [magnars/multiple-cursors.el#318]
I am not really good in elisp but I hacked together a function that

``` emacs-lisp
  (defun mark-word-or-next-word-like-this ()
    "if there is no active region the word under
     the point will be marked, otherwise the next word is selected."
    (interactive)
    (if (region-active-p)
    ;; then
      (progn
        (mc/mark-more-like-this nil 'forwards)
        (mc/maybe-multiple-cursors-mode)
        (mc/cycle-forward))
    ;; else
      (mc--select-thing-at-point 'word)))
```

So like the normal `mc/mark-more-like-this`, when reaching the end of the buffer no more matches will be found. Is there a way to start searching from the top, if the bottom is reached?

**Comment by rksm** [comment]
There is `mc/cycle-looping-behaviour` which can have any of the values:

```
'continue "Loop around to beginning/end of document."
'warn     "Warn and then loop around."
'error    "Signal an error."
'stop     "Don't loop."
```

**Comment by FelixBrendel** [comment]
The problem is not the cycling, but rather not finding the next word-like-this when there are no more matches until the end of the buffer.
Basically I want to replicate Atoms behaviour:
![atom](https://user-images.githubusercontent.com/5005975/35142848-f0cf973a-fcff-11e7-842b-2faa534c3a59.gif)
In emacs so far I have 
![emacs](https://user-images.githubusercontent.com/5005975/35142854-f6f6a572-fcff-11e7-8f01-9bf3101b5146.gif)
Sorry If it wasn't clear before!

**Comment by nispio** [comment]
I don't think that this is currently possible.  `multiple-cursors-mode` uses the function `re-search-forward` to find the next match, and I don't see anyway to enable wrapping for that function.  I think that `multiple-cursors-mode` would require modifications in order to make this work smoothly.

I think that it would be worthwhile to add this ability to `mc/mark-more-like-this` and control the behavior with an option very similar to `mc/cycle-looping-behavior`.  Any volunteers?

**Comment by Compro-Prasad** [comment]
@jtamagnan I did zero config and it works well for `mc/mark-next-like-this`. How do I restrict it to wrap around if I needed in some case?
